src/backend/distributed/commands/schema_based_sharding.c

```diff
@@ -625,6 +625,23 @@ citus_schema_distribute(PG_FUNCTION_ARGS)
 
 	ereport(NOTICE, (errmsg("distributing the schema %s", schemaName)));
 
+	/* Propagate the schema if there is no table in the schema yet */
+	ObjectAddress *schemaAddress = palloc0(sizeof(ObjectAddress));
+	ObjectAddressSet(*schemaAddress, NamespaceRelationId, schemaId);
+	if (tableIdListInSchema == NIL && EnableMetadataSync)
```

Comment:
is this different from EnsureDependenciesExistOnAllNodes ?

Reply:
Yes. `EnsureDependenciesExistOnAllNodes` does not create the object itself but creates dependencies. The reasons we should propagate inside `DistributeSchema` are 

1. If we propagate the schema in postprocess and mark it as distributed, then outside transaction could not see it (visibility problem),
2. If we propagate the schema in postprocess without marking it as distributed, then outside transaction is blocked by coordinated transaction for the propagation.

We can always execute idempotent schema propagation in coordinated transaction after converting all tables under the schema. There could be 2 situations which both are ok:

1. No tables under the schema. => the schema would not be created by `EnsureDependenciesExistOnAllNodes`. But we would propagate it afterwards,
2. There is at least 1 table under the schema => the schema would be created by `EnsureDependenciesExistOnAllNodes`. Then, propagation of the schema would be a no-op. (idempotent)

Reply:
Now the problem is the below fails during propagation since user does not have create privilage on database even if the user is the owner of the schema.

1. Create the schema with privileged user,
2. Pass table ownership to unprivileged user,
3. Propagation during schema distribution fails since the user cannot create schema on database.

```sql
CREATE SCHEMA <schema> AUTHORIZATION <user_with_no_create_on_database>
```

Still thinking the proper way. It may be little bit weird to enforce `CREATE ON DATABASE` privilege in addition to table owner check since the schema already exists.

Reply:
Actually this looks like an expected case even if we propagate at postprocess. The problem is that we do not propagate grants, so it is expected to get error from workers in those cases.